src/affinder/affinder.py

```diff
@@ -82,7 +82,44 @@ def close_affinder(layers, callback):
         layer.mode = 'pan_zoom'
 
 
+def _update_unique_choices(widget, choice_name):
+    """Update the selected choice in a ComboBox widget to be unique.
+
+    When `choice_name` is selected by another widget, and the choice in
+    `widget` needs to be different, this callback can be called to update the
+    choice in `widget`.
+    """
+    if not isinstance(choice_name, str):
+        # in some circumstances, widget.changed.connect passes the choice
+        # name to the callback, and in other cases it's the actual choice
+        # value. Here we coerce it to always be the name but that's an
+        # arbitrary choice.
+        choice_name = choice_name.name
+    choices = widget.choices
+    choice_names = [value.name for value in choices]
+    index = choice_names.index(choice_name)
+    value = widget.choices[index]
+    if widget.value is value:
+        next_index = (index+1) % len(choices)
+        widget.changed.block()
+        # should block() be a context manager?
+        widget.value = widget.choices[next_index]
+        widget.changed.unblock()
```

Comment:
@tlambert03 am I doing it right here? Feels icky. 😂

Reply:
is the question whether there is a context manager?  if so, yes: https://psygnal.readthedocs.io/en/latest/API/#psygnal._group.SignalGroup.blocked

Reply:
LOL Yes that was the question! When tabbing through the API I thought `blocked` would be a boolean for whether it's blocked. What do you think about making `block` return a context manager? I understand where you're coming from from a grammatical perspective, but I think most context managers tend to use present tense? e.g. `with open(path) as fin:`, `with suppress(Exception):`, `with catch_warnings()`, `with mock.patch`...

Reply:
Yeah we've been down this conversation before.  I know the open example, but I don't think it's the "only" pattern in the standard library and I prefer grammatical adjectives (as opposed to verbs) for context managers.  And I'd like to be consistent with myself :)

Reply:
I think the docs are sufficient to find it ;)

Reply:
In fact, there's even a suggestion in pep https://peps.python.org/pep-0343/#examples (though clearly not internally adopted everywhere):

> The tense used in the names of the example contexts is not arbitrary. Past tense (“-ed”) is used when the name refers to an action which is done in the __enter__ method and undone in the __exit__ method. Progressive tense (“-ing”) is used when the name refers to an action which is to be done in the __exit__ method.

Reply:
Nice, ok. Good enough for me. 🙏 Very annoying that the stdlib doesn't follow their own recommendation LOL

Reply:
I think open() actually predates the with keyword, for what it's worth.

You can see guido even creates an `opened` function that wraps `open` in that example.

Reply:
Yeah if it was only "open" I wouldn't have mentioned anything. It's all of contextlib that bugs me. LOL.